src/shogun/features/DenseFeatures.h

```diff
@@ -541,6 +543,23 @@ template<class ST> class CDenseFeatures: public CDotFeatures
 	void init();
 
 protected:
+	/*
+	 * Helper method which copies the working feature matrix into the pre-allocated
+	 * target matrix passed to this method. If the size of the pre-allocated matrix is
+	 * not sufficient to copy all the feature vectors, it reallocates the memory to fit
+	 * the required size, with the column_offset. It then copies into the target
+	 * matrix, starting from base + (colum_offset * num_features) location.
+	 *
+	 * In case reallocation happens, it initializes the memory till the offset location
+	 * with 0, in order to avoid leaving it uninitialized. However, if reallocation
+	 * does not happen within this method, it is assumed that it is the responsilibity
+	 * of the caller to initialize the offsetted part.
+	 *
+	 * Please note that this method always does a deep copy, irrespective of whether
```

Comment:
What I mean is, even if there are no subsets, it still creates a new SGMatrix and copies into it rather than returning the original one. But I agree, only copy would suffice to convey the same sense :)